services/supervisor/src/api.rs

```diff
@@ -93,6 +96,15 @@ impl PrivateApi for ApiImpl<'_> {
 
         Ok(configuration_number)
     }
+
+    fn supervisor_config(&self) -> Result<SupervisorConfig, Self::Error> {
+        let config = Schema::new(self.0.service_data())
+            .configuration
+            .get()
+            .expect("Supervisor entity was not configured; unable to load configuration");
```

Comment:
Why not to return `Err` ? Do we really need to panic here ?

Reply:
Yes. Comment from `lib.rs`:
```
            // Incorrect config is a critical error for both the service and the
            // blockchain itself: supervisor can't operate not configured, and the
            // blockchain isn't expected to work without supervisor.
```

Achieving such a condition means a critical and unrecoverable error.

Reply:
Ok. But what about user ? :) Why don't answer him about this case ? If user makes REST API request and there is no config... so as I understand he never knows about it because a node will crash, right ? IMO it's not good from user experience side.

Reply:
Well, look: this condition isn't achievable in any imaginable scenario -- we require config to be added within initialization and we validate that it can be parsed. There is no valid way to remove the config that once been set (it's only possible to change it to other valid config). So, if, for some reason, node with supervisor have started successfully, but the config was not set, it means that something gone terribly wrong, and the UX is the least of all possible problems at the moment.

Reply:
In other words, this `except` is in fact completely safe. Scenario when this check will panic is (almost) impossible, and should never happen.